cvlib/src/corner_detector.cpp

```diff
@@ -16,10 +16,89 @@ cv::Ptr<corner_detector_fast> corner_detector_fast::create()
     return cv::makePtr<corner_detector_fast>();
 }
 
-void corner_detector_fast::detect(cv::InputArray image, CV_OUT std::vector<cv::KeyPoint>& keypoints, cv::InputArray /*mask = cv::noArray()*/)
+bool corner_detector_fast::checkPixel(cv::Mat &image, int i, int j, int N, int t)
+{
+    std::vector<int> _successVerify = std::vector<int>(16, -255);
+    int count_black = 0;
+    int count_white = 0;
+
+    for (auto firstIndex : _initialVerify)
+    {
+
+        if (image.at<uchar>(cv::Point(j, i) + _pixelsAround[firstIndex - 1]) > image.at<uchar>(i, j) + t)
+        {
+            count_white++;
+            _successVerify[firstIndex - 1] = 255;
+        }
+        else if (image.at<uchar>(cv::Point(j, i) + _pixelsAround[firstIndex - 1]) < image.at<uchar>(i, j) - t)
+        {
+            count_black++;
+            _successVerify[firstIndex - 1] = 0;
+        }
+    }
+
+    if ((count_black >= radius) || (count_white >= radius))
+    {
+        for (auto secondIndex : _secondaryVerify)
+        {
+            if (image.at<uchar>(cv::Point(j, i) + _pixelsAround[secondIndex - 1]) > image.at<uchar>(i, j) + t)
+            {
+                _successVerify[secondIndex - 1] = 255;
+            }
+            else if (image.at<uchar>(cv::Point(j, i) + _pixelsAround[secondIndex - 1]) < image.at<uchar>(i, j) - t)
+            {
+                _successVerify[secondIndex - 1] = 0;
+            }
+        }
+
+        if (checkMaxLenSeqPix(_successVerify, N))
+            return true;
+    }
+
+        return false;
+}
+
+bool corner_detector_fast::checkMaxLenSeqPix(std::vector<int> seq, int N)
+{
+    int countMax = 1;
+    int count = 1;
+
+    for (size_t k = 1; k < seq.size(); k++)
```

Comment:
Ещё нужно учесть, что точки расположены по кольцу, т.е. поиск одинаковых элементов тоже нужно "замкнуть". Например, может быть такая ситуация: `0 0 0 0 1 0 1 0 1 0 0 0`. В этой последовательности 7 нулевых элементов подряд.

Reply:
Это можно решить таким вот способом:
```suggestion
    f    int countMax = 1;
    int count = 1;
    int size = seq.size();

    for (size_t k = 1; k < 2*size - 2; k++)
    {
        if (seq[k%size] == seq[(k - 1)%size])
            count++;
        else
            count = 1;

        if (count > countMax)
            countMax = count;
    }
```